tekton/pipeline_run.go

```diff
@@ -66,6 +68,13 @@ func CreateReleasePipelineRun(release *v1alpha1.Release, releaseStrategy *v1alph
 				Name:   releaseStrategy.Spec.Pipeline,
 				Bundle: releaseStrategy.Spec.Bundle,
 			},
+			Params: []tektonv1beta1.Param{{ // This is silently ignored if pipeline called doesn't declare this parameter
```

Comment:
If we go through with my PR, this could be done with something like the following:

```
func (pr *ReleasePipelineRun) WithApplicationSnapshot(snapshot *v1alpha1.ApplicationSnapshot) *ReleasePipelineRun {
	snapshotString, err := json.Marshal(snapshot.Spec)
	if err != nil {
		return nil
	}

	pr.Spec.Params = append(pr.Spec.Params, tektonv1beta1.Param{
		Name: snapshot.Kind,
		Value: tektonv1beta1.ArrayOrString{
			Type:      tektonv1beta1.ParamTypeString,
			StringVal: string(snapshotString),
		},
	})

	return pr
}
```

Reply:
Yes, will wait to base this PR on that one